services/templates/service.go

```diff
@@ -9,3 +29,65 @@ type service struct {
 func NewTemplateService() Service {
 	return &service{}
 }
+
+func (s service) AddTemplate(c context.Context, input *entities.Template) error {
+
+	// parse string to validate template params
+	_, err := mustache.ParseString(input.HTMLPart)
+	if err != nil {
+		return ErrParseHTMLPart
+	}
+	// parse string to validate template params
+	_, err = mustache.ParseString(input.TextPart)
+	if err != nil {
+		return ErrParseTextPart
+	}
+	// parse string to validate template params
+	_, err = mustache.ParseString(input.SubjectPart)
+	if err != nil {
+		return ErrParseSubjectPart
+	}
+
+	err = storage.CreateTemplate(c, input)
+	if err != nil {
+		return fmt.Errorf("failed to create template error: %w", err)
```

Comment:
Errors should be shorter, something like this:

"create template: %w"

And

"save template to s3: %w"